### PR TITLE
style: enable revive linter

### PR DESCRIPTION
**Pull request by bernot-dev**
Revive is a meta-linter with many available rules, fixes were made in accordance with these default rules:

- empty-block
- error-strings
- increment-decrement
- indent-error-flow
- receiver-naming
- unexported-return
- unused-parameter
- var-naming

See [Revive Available Rules](https://github.com/mgechev/revive?tab=readme-ov-file#available-rules) for details.

Then, enabled all revive rules. Disabled all currently failing rules, to be evaluated on a case-by-case basis for future inclusion.

### CODE REVIEW

File: cmd/frontend/main.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func main() {
 			//nolint:errcheck
```
Comment (macxamin) [inline cmd/frontend/main.go:153]: Is there a plan to remove these?

**Review by macxamin — APPROVED** [review]
LGTM

File: .golangci.yml [hunk 2]
```diff
@@ -84,7 +83,150 @@ linters-settings:
...
+        severity: warning
+        disabled: true
+      # https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#useless-break
+      - name: useless-break
+        severity: warning
+        disabled: true
 issues:
+  exclude-use-default: false
+  exclude:
+    # errcheck: Almost all programs ignore errors on these functions and in most cases it's ok
```
Comment (macxamin) [inline .golangci.yml:228]:
```suggestion
    # errcheck: Almost all programs ignore errors on these functions and in most cases it's ok.
```

File: cmd/frontend/main.go (hunk 1 above)
Comment (bernot-dev) [reply cmd/frontend/main.go:153]: This is a weird one. The logger can technically return an error, which is not being checked. Whether that's a useful error check is debatable. So, we have a few options:

1. Check for errors on logging, when using this logger.
2. Leave the `//nolint` annotations in place.
3. Add an exception to recognize this pattern of error and ignore it.
4. Use a different logging library that doesn't return errors.

Not sure which I would prefer yet, personally.

**Pull request merged**